src/pip/_internal/metadata/base.py

```diff
@@ -131,6 +145,26 @@ def editable_project_location(self) -> Optional[str]:
                 return self.location
         return None
 
+    @property
+    def installed_location(self) -> Optional[str]:
+        """The distribution's "installed" location.
+
+        This should generally be a ``site-packages`` directory. This is
+        usually ``dist.location``, except for legacy develop-installed packages,
+        where ``dist.location`` is the source code location, and this is where
+        the ``.egg-link`` file is.
+
+        The returned location is normalized (in particular, with symlinks removed).
+        """
+        egg_link = egg_link_path_from_location(self.raw_name)
+        if egg_link:
+            location = egg_link
+        elif self.location:
+            location = self.location
+        else:
+            return None
+        return str(pathlib.Path(location).resolve())
```

Comment:
Why not use `_normalize_path` here?

Reply:
Good catch, I forgot 🙂 (I was doing this repeatedly in the beginning and decided to create `_normalize_path` only later.)